packages/react-components/src/components/chart/hooks/useTrendCursorsEvents.ts

```diff
@@ -151,17 +172,46 @@ const useTrendCursorsEvents = ({
         });
       }
 
+      chart?.getZr().on('contextmenu', (event) => {
+        onContextMenuRef.current(event);
+      });
+
       return () => {
         if (ref.current !== null) {
           chart?.getZr().off('click');
           // passing the function so that it will NOT remove all the mouseover behaviour, namely default tooltip
           chart?.getZr().off('mouseover', () => mouseoverHandler(isInCursorAddModeRef.current, chart));
           chart?.getZr().off('drag');
+          chart?.getZr().off('contextmenu');
         }
       };
     }
     prevRef.current = ref;
   }, [ref]);
+
+  const copyTrendCursorData = (posX: number) => {
+    const timestampOfClick = calculateTimeStamp(posX, sizeRef.current.width, viewportRef.current);
+    const toBeCopiedGraphicIndex = calculateNearestTcIndex(graphicRef.current, timestampOfClick);
+    // using copy-to-clipboard library to copy in a Excel sheet pastable format
+    copy(formatCopyData(graphicRef.current[toBeCopiedGraphicIndex], seriesRef.current), { format: 'text/plain' });
+  };
+
+  // this handles the user interaction via context menu
+  const onContextMenuClickHandler = ({ action, posX }: { action: Action; posX: number }) => {
+    switch (action) {
```

Comment:
Without a default, what happens if a fourth action is added and this switch statement isn't updated?

Reply:
No action occurs, did not want to have any action if by chance someone changes the Action type